pkg/v1/webhooks/certificates.go

```diff
@@ -0,0 +1,131 @@
+func AddCertToValidatingWebhookConfig(ctx context.Context, client kubernetes.Interface, labelSelector string, secret *corev1.Secret) error {
+	allValidatingWebhookConfigurations, err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().List(ctx, metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return err
+	}
+	for idx := range allValidatingWebhookConfigurations.Items {
+		configName := allValidatingWebhookConfigurations.Items[idx].Name
+		validatingWebhookConfiguration, err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Get(ctx, configName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		for idx := range validatingWebhookConfiguration.Webhooks {
+			validatingWebhookConfiguration.Webhooks[idx].ClientConfig.CABundle = secret.Data[resources.CACert]
+		}
+		if _, err := client.AdmissionregistrationV1().ValidatingWebhookConfigurations().Update(ctx, validatingWebhookConfiguration, metav1.UpdateOptions{}); err != nil {
+			return fmt.Errorf("error updating CA cert of ValidatingWebhookConfiguration %s: %v", configName, err)
+		}
+	}
+	return nil
+}
+
+//nolint:dupl
+func AddCertToMutatingWebhookConfig(ctx context.Context, client kubernetes.Interface, labelSelector string, secret *corev1.Secret) error {
+	allMutatingWebhookConfigurations, err := client.AdmissionregistrationV1().MutatingWebhookConfigurations().List(ctx, metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return err
+	}
+	for idx := range allMutatingWebhookConfigurations.Items {
+		configName := allMutatingWebhookConfigurations.Items[idx].Name
+		mutatingWebhookConfiguration, err := client.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(ctx, configName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		for idx := range mutatingWebhookConfiguration.Webhooks {
+			mutatingWebhookConfiguration.Webhooks[idx].ClientConfig.CABundle = secret.Data[resources.CACert]
+		}
+		if _, err := client.AdmissionregistrationV1().MutatingWebhookConfigurations().Update(ctx, mutatingWebhookConfiguration, metav1.UpdateOptions{}); err != nil {
+			return fmt.Errorf("error updating CA cert of MutatingWebhookConfiguration %s: %v", configName, err)
+		}
+	}
+	return nil
+}
+
+func UpdateTLSSecret(ctx context.Context, secretName, serviceName, certPath, keyPath, namespace string) (*corev1.Secret, error) {
+	secret, err := resources.MakeSecretInternal(ctx, secretName, namespace, serviceName)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := certutil.WriteCert(certPath, secret.Data[resources.ServerCert]); err != nil {
+		return secret, err
+	}
+	if err := keyutil.WriteKey(keyPath, secret.Data[resources.ServerKey]); err != nil {
+		return secret, err
+	}
+
+	return secret, nil
+}
+
+func UpdateCertificates(ctx context.Context, k8sConfig *rest.Config, certPath, keyPath, secretName, namespace, serviceName, labelSelector string) (*corev1.Secret, error) {
```

Comment:
Should other webhooks test files call this function to update the certificate before running the test cases?

Reply:
yes.   This will write the certificates to the provided paths. In reallity this exists independent of the webhook server. 
What it basically does is write the tls.cert and tls.keys to the given directories, and inject  the CA bundle into the webhook configs. 
 so the webhook server does not need to even be running. It just hast to be configured with the matching certDir

you basically want certPath = certDir + "tls.cert" and keypath = CertDir + "tls.key"
Then when this gets run the cerfiticates are "installed" and the webhook will automatically pick it up (if it is running already) or will pick it up on start up. 

The CertDir needs to be created already. 

So it should be something like

install webhook configuration (manfiests.yml)
create  certdir ( to prevent racing condition when webhook takes too long to start)
run this function (changed name from UpdateCertificates to "InstallNewCertificates" 
configure webhookserver with certdir and start it
then ready to run test. 

The order could be change a bit, just need to make sure the CertDir is available before you run this function.